File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Foundation/FunctionApplicationIterator.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using System.Collections;
using Db4objects.Db4o.Foundation;

namespace Db4objects.Db4o.Foundation
{
	/// <exclude></exclude>
	public class FunctionApplicationIterator : MappingIterator
	{
		private readonly IFunction4 _function;

		public FunctionApplicationIterator(IEnumerator iterator, IFunction4 function) : base
			(iterator)
		{
			if (null == function)
			{
				throw new ArgumentNullException();
			}
			_function = function;
		}

		protected override object Map(object current)
		{
			return _function.Apply(current);
		}
	}
}

File: db4oj/core/src/com/db4o/foundation/FunctionApplicationIterator.java
/* Copyright (C) 2006 Versant Inc. http://www.db4o.com */

package com.db4o.foundation;

/**
 * @exclude
 */
public class FunctionApplicationIterator extends MappingIterator {

	private final Function4 _function;
	
	public FunctionApplicationIterator(Iterator4 iterator, Function4 function) {
		super(iterator);
		if (null == function) {
			throw new ArgumentNullException();
		}
		_function = function;
	}
	
	protected Object map(final Object current) {
		return _function.apply(current);
	}	
}
